.github/workflows/ci.yaml

```diff
@@ -0,0 +1,39 @@
+name: Continuous Integration
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build_and_test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.12'
+
+      - name: Install Poetry
+        run: |
+          export POETRY_HOME=/opt/poetry
+          curl -sSL https://install.python-poetry.org | python3 -
+          echo "$POETRY_HOME/bin" >> $GITHUB_PATH
```

Comment:
Is there a reason to do this (quite insecure) direct download instead of a regular `apt install python3-poetry` ?

Reply:
No reason in particular it was just one of the ways suggested  [here](https://python-poetry.org/docs/#ci-recommendations) in the CI recommendations section of the Python-Poetry docs.

Reply:
I didn't think of it being insecure but I will switch then to the `apt install python3-poetry` solution!

Reply:
Finally, the version that was working with no error was with `pipx install poetry`, which is also one of the suggested solutions in the [Python-Poetry docs](https://python-poetry.org/docs/#ci-recommendations).